h5py/h5g.pyx

```diff
@@ -389,9 +394,9 @@ cdef class GroupID(ObjectID):
         if statbuf.type != H5G_LINK:
             raise ValueError('"%s" is not a symbolic link.' % name)
 
-        IF UNAME_SYSNAME == "Windows":
+        if _IS_WINDOWS:
             linklen = 2049  # Windows statbuf.linklen seems broken
```

Comment:
I did a rapid scan of the exact (massive) commit this line was introduced and didn't find any related-looking test, and I don't know how else I could be looking for it.

Reply:
I'm thinking we should make sure there's a test that runs this function (either directly or through the high-level API), and ideally includes a link value over 2048 bytes long.  Then get rid of the windows special case and see if those tests can pass on Windows.

Reply:
I did the easy thing first: I just tried removing this branch and ran all tests on windows, and got no error. So at least we know we're not in the scenario where we have a test *and* this workaround is still needed.
Now it might be easier to craft a brand new test to exercise the condition than to discover (or prove lack of) an existing one. I'm not sure I would know how to do that but at this point I'm rather convinced it's a task on its own and should be done in a separate PR.